packages/core/src/sdk/DatadogProvider/Buffer/BoundedBuffer.ts

```diff
@@ -164,6 +155,23 @@ export class BoundedBuffer extends DatadogBuffer {
         this.drainTelemetry();
     };
 
+    private logBufferOverflow() {
+        this.addTelemetryEvent('Buffer overflow', '', 'BufferOverflow');
+    }
+
+    private logRandomIdGenerationError = (error: any): void => {
+        this.addTelemetryEvent(
+            'Could not generate enough random numbers',
+            getErrorStackTrace(error),
+            'RandomIdGenerationError'
+        );
+
+        // Not using InternalLog here as it is not yet instantiated
+        console.warn(
+            `[Datadog] Could not generate enough random numbers for RUM buffer. Please check that Math.random is not overwritten. Math.random returns: ${Math.random()}`
+        );
+    }
+
```

Comment:
This is just a small refactoring I have made in a previous iteration, and I have decided to keep it in this PR.